txpool/tx_object_test.go

```diff
@@ -29,25 +28,21 @@ func newChainRepo(db *muxdb.MuxDB) *chain.Repository {
 	return repo
 }
 
-func signTx(tx *tx.Transaction, acc genesis.DevAccount) *tx.Transaction {
-	sig, _ := crypto.Sign(tx.SigningHash().Bytes(), acc.PrivateKey)
-	return tx.WithSignature(sig)
-}
-
 func newTx(chainTag byte, clauses []*tx.Clause, gas uint64, blockRef tx.BlockRef, expiration uint32, dependsOn *thor.Bytes32, features tx.Features, from genesis.DevAccount) *tx.Transaction {
 	builder := new(tx.Builder).ChainTag(chainTag)
 	for _, c := range clauses {
 		builder.Clause(c)
 	}
 
-	tx := builder.BlockRef(blockRef).
+	tx, _ := builder.BlockRef(blockRef).
 		Expiration(expiration).
 		Nonce(rand.Uint64()). // nolint:gosec
 		DependsOn(dependsOn).
 		Features(features).
-		Gas(gas).Build()
+		Gas(gas).
+		BuildAndSign(from.PrivateKey)
```

Comment:
Can we add here too the `require.NoError(t, err)` statement?

Reply:
Used a MustSign